### PR TITLE
feat(signature-collection-api-domains): Granular access

### PR DESCRIPTION
**Pull request by kksteini**
## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced new decorators: `AllowDelegation`, `IsOwner`, `AllowManager`, and `RestrictGuarantor` for enhanced access control.
	- Added a centralized index for decorators and data transfer objects related to signature collections.
	- Implemented a `MetadataAbstractor` class for streamlined metadata retrieval.

- **Bug Fixes**
	- Removed outdated access requirement decorators, simplifying access control logic.

- **Tests**
	- Added comprehensive unit tests for the `UserAccessGuard`, validating access control across various user roles.

- **Chores**
	- Updated TypeScript configuration for improved compiler behavior.

### CODE REVIEW

File: libs/api/domains/signature-collection/src/lib/signatureCollection.resolver.ts
Comment (juni-haukur) [inline libs/api/domains/signature-collection/src/lib/signatureCollection.resolver.ts:1]: Á ekki IsOwner decoratorinn að vera bara þar sem gamla @AccessRequirement(OwnerAccess.XXXX) var?

File: libs/api/domains/signature-collection/src/lib/guards/userAccess.guard.ts [hunk 1]
```diff
@@ -24,36 +20,41 @@ export class UserAccessGuard implements CanActivate {
...
+    }
+
+    const isDelegationRestrictions = this.reflector.getAllAndOverride<
+      UserDelegationContext[] | undefined
+    >('delegation-requirement', [context.getHandler(), context.getClass()])
+
+    const allowsDelegationToOwner = isDelegationRestrictions?.some(
+      (t) =>
+        t === UserDelegationContext.PersonDelegatedToPerson ||
+        t === UserDelegationContext.PersonDelegatedToCompany,
```
Comment (juni-haukur) [inline libs/api/domains/signature-collection/src/lib/guards/userAccess.guard.ts:41]: PersonDelegatedToPerson og PersonDelegatedToCompany er hvergi sett í resolvernum - bara AllowAll þegar umboð má framkvæma aðgerð

**Review by juni-haukur — COMMENTED** [review]
Meikar sens að splitta upp delegationRequirement til að samræma við MMK contextið?
t.d. fyrir framboðs-endapunkta: @CandidacyRequirement(ábyrgðar / umsjónar) 
og user-endapunkta þar sem þarf að restricta actors: @ActorRestriction()

**Review by juni-haukur — APPROVED** [review]
Approved - með fyrirvara um 1x IsOwner comment

File: libs/api/domains/signature-collection/src/lib/signatureCollection.resolver.ts [hunk 2]
```diff
@@ -99,7 +102,7 @@ export class SignatureCollectionResolver {
   }
 
   @Scopes(ApiScope.signatureCollection)
-  @AccessRequirement(UserAccess.RestrictActor)
+  @IsOwner()
```
[the remark this answers is not in the document]
Comment (kksteini) [reply libs/api/domains/signature-collection/src/lib/signatureCollection.resolver.ts:105]: Vel spottað!

File: libs/api/domains/signature-collection/src/lib/guards/userAccess.guard.spec.ts [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+  })
+
+  it('Should allow owner to access IsOwner decorated paths', async () => {
+    setupMockForUser(userIsOwnerNotDelegated)
+
+    const response = await request(app.getHttpServer())
+      .get('/graphql')
+      .query({ query: '{ getIfOwner }' })
+
+    expect(response.body).toMatchObject(okGraphQLResponse('getIfOwner'))
```
[the remark this answers is not in the document]
Comment (kksteini) [reply libs/api/domains/signature-collection/src/lib/guards/userAccess.guard.spec.ts]: Good suggestion

**Comment by kksteini** [comment]
@coderabbitai review

**Pull request merged**